#!/usr/bin/env python
PACKAGE = "voxel_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen=ParameterGenerator()
gen.add('input_cloud',str_t,0,"input_cloud","/left_os1/os1_cloud_node/points")
gen.add('max_x',double_t,0,"Max_x",100,-100.0,100)
gen.add('min_x',double_t,0,"Min_x",-4.0,-100,100)
gen.add('max_y',double_t,0,"Max_y",12.0,-50,50)
gen.add('min_y',double_t,0,"Min_y",-12.0,-50,50)
gen.add('max_z',double_t,0,"Max_z",-0.8,-10,10)
gen.add('min_z',double_t,0,"Min_z",-1.8,-10,10)
gen.add('useVoxelF',bool_t,0,"Use voxel filtering",False)
gen.add('lx',double_t,0,"leaf x",0.2,0,5)
gen.add('ly',double_t,0,"leaf y",0.2,0,5)
gen.add('lz',double_t,0,"leaf z",0.2,0,5)

exit(gen.generate(PACKAGE, "voxel_filter", "filter_node"))